### ISSUE TITLE
[CL][Swap Router Module]: implement store index to map pool id to swap module 

### ISSUE THREAD

**Issue by p0mvn** [osmosis-labs/osmosis#3238]
## Background

We need to store a mapping from pool id to swap module (`gamm` or `concentrated-liquidity`).

This is necessary for swaps. Given the pool id, we should be able to know which module to route the swap to.

Currently, `swaprouter` hardcodes all swaps to be routed to `gamm`.

## Suggested Design

See for details: https://github.com/osmosis-labs/osmosis/pull/3233

1. Implement:

```go
// x/swaprouter/types/keys.go
var	SwapModuleRouterPrefix     = []byte{0x01}
type ModuleRoute struct {
    PoolId uint64
    PoolType PoolType
}
// ModuleRouteToBytes serializes moduleRoute to bytes.
func ModuleRouteToBytes(moduleRoute ModuleRoute) []byte {
    ...
}
// ModuleRouteFromBytes deserializes an encoded module route. It returns
// an error if decoding fails.
func ModuleRouteFromBytes(bz []byte) (ModuleRoute, error) {
    ...
}
```

2. Use the above to store the mapping in `CreatePool`
- update tests to check that the store is updated

3. Implement pseudocode.

```go
// x/swaprouter/router.go RouteExactAmountIn(...)
moduleRouteBytes := osmoutils.MustGet(swaproutertypes.FormatModuleRouteIndex(poolId))
moduleRoute, _ := swaproutertypes.ModuleRouteFromBytes(moduleRouteBytes)
swapModule := k.routes[moduleRoute.PoolType]
_ := swapModule.SwapExactAmountIn(...)
```

4. Optionally in the same PR, complete: https://github.com/osmosis-labs/osmosis/issues/3236

5. Optionally in the same PR: https://github.com/osmosis-labs/osmosis/issues/3131

## Acceptance Criteria
- suggested design implemented
- proper abstractions added, not just pseudocode
- requested tests added

**Comment by p0mvn** [comment]
Before proceeding, consider alternatives from: https://github.com/osmosis-labs/osmosis/issues/3097

**Comment by p0mvn** [comment]
fixed by #3097
